sources/packages/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -32,88 +28,45 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
-      .select("restrictions.id", "restrictionId")
-      .addSelect("restrictions.restrictionType", "restrictionType")
-      .addSelect("count(*)", "restrictionCount")
-      .innerJoin("studentRestrictions.restriction", "restrictions")
-      .innerJoin("studentRestrictions.student", "student")
-      .innerJoin("student.user", "user")
-      .where("studentRestrictions.isActive = true")
-      .andWhere("user.id= :userId", {
-        userId,
-      })
-      .groupBy("studentRestrictions.student.id")
-      .addGroupBy("restrictions.id")
-      .addGroupBy("restrictions.restrictionType")
-      .getRawMany();
-
-    if (!result || result.length === 0) {
-      return {
-        hasRestriction: false,
-        hasFederalRestriction: false,
-        hasProvincialRestriction: false,
-        restrictionMessage: null,
-      } as StudentRestrictionStatus;
-    }
-    let restrictionMessage: string = null;
-    const hasFederalRestriction = result.some(
-      (item) => item.restrictionType === RestrictionType.Federal.toString(),
-    );
-
-    const hasProvincialRestriction = result.some(
-      (item) => item.restrictionType === RestrictionType.Provincial.toString(),
-    );
-
-    if (hasFederalRestriction) {
-      restrictionMessage = RESTRICTION_FEDERAL_MESSAGE;
-    }
-
-    if (hasProvincialRestriction) {
-      restrictionMessage = restrictionMessage
-        ? restrictionMessage + " " + RESTRICTION_PROVINCIAL_MESSAGE
-        : RESTRICTION_PROVINCIAL_MESSAGE;
-    }
-
-    return {
-      hasRestriction: true,
-      hasFederalRestriction: hasFederalRestriction,
-      hasProvincialRestriction: hasProvincialRestriction,
-      restrictionMessage: restrictionMessage,
-    } as StudentRestrictionStatus;
-  }
-
   /**
    * Creates a 'select' query that could be used in an 'exists' or
    * 'not exists' where clause to define if the student has some
    * active restrictions that must prevent him for certain
    * critical operations, for instance, to have money disbursed.
    * ! This query will assume that a join to 'student.id' is present
    * ! in the master query.
-   * TODO: Removed .having("count(*) > restriction.allowedCount"), which may cause unexpected results, need to adjust the logic.
+   * @param restrictionActions list of restriction actions that needed to be checked.
+   * @param checkAll, is by default false, checkAll decides if the all the
+   * elements in the restrictionActions should be checked (i.e checkAll = true)
+   * or any one of the element is to be checked (i.e checkAll = false).
    * @returns 'select' query that could be used in an 'exists' or
    * 'not exists'.
    */
-  getExistsBlockRestrictionQuery(): SelectQueryBuilder<StudentRestriction> {
-    return this.repo
+  getExistsBlockRestrictionQuery(
+    restrictionActions: RestrictionActionType[],
+    checkAll = false,
+  ): SelectQueryBuilder<StudentRestriction> {
+    const formatArrayForQuery = JSON.stringify(restrictionActions).replace(
```

Comment:
Why are we doing this replace?

Reply:
here array comparison is done inside the sub query and it except the array with elements in single quotes. JSON. stringify returns the Array string in double-quotes.

Reply:
@ann-aot, apparently there are some alternatives, did you try something like the below? Please note that the check is inverted here.
![image](https://user-images.githubusercontent.com/61259237/170589362-2afc0e83-6a94-4b36-92f1-3ab3e681ff29.png)
https://wanago.io/2020/11/02/api-nestjs-array-data-type-postgresql-typeorm/

Reply:
It will be nice to put a comment here.